esphome/components/online_image/png_image.cpp

```diff
@@ -38,6 +39,14 @@ static void draw_callback(pngle_t *pngle, uint32_t x, uint32_t y, uint32_t w, ui
   PngDecoder *decoder = (PngDecoder *) pngle_get_user_data(pngle);
   Color color(rgba[0], rgba[1], rgba[2], rgba[3]);
   decoder->draw(x, y, w, h, color);
+
+  // Feed watchdog periodically to avoid triggering during long decode operations.
+  // Feed every 1000 pixels to balance efficiency and responsiveness.
+  uint32_t pixels = w * h;
+  decoder->add_pixels(pixels);
+  if ((decoder->get_pixels_decoded() % 1000) < pixels) {
```

Comment:
The suggestion from @swoboda1337 to use 1024 instead sound good though!
```suggestion
  if ((decoder->get_pixels_decoded() % 1024) < pixels) {
```